bin/templates/scripts/cordova/lib/run.js

```diff
@@ -147,25 +146,42 @@ function filterSupportedArgs (args) {
  * Checks if any iOS device is connected
  * @return {Promise} Fullfilled when any device is connected, rejected otherwise
  */
-function checkDeviceConnected () {
-    return superspawn.spawn('ios-deploy', ['-c', '-t', '1'], { printCommand: true, stdio: 'inherit' });
+function checkDeviceConnected() {
+    return Q.Promise(function (resolve, reject) {
+        return iosDevice.devices(function (err, devices) {
+            if (err || !devices.length) {
+                return reject(err || new Error('no device connected'));
+            }
+            return resolve(devices[0].udid);
+        });
+    })
 }
 
 /**
  * Deploy specified app package to connected device
- * using ios-deploy command
+ * using node-ios-device
  * @param  {String} appPath Path to application package
  * @return {Promise}        Resolves when deploy succeeds otherwise rejects
  */
-function deployToDevice (appPath, target, extraArgs) {
```

Comment:
What was `extraArgs` used for?

Reply:
`extraArgs` contained the arguments not used by `run.js`.
It allowed to pass arguments to `ios-deploy`([ios-deploy usage](https://github.com/ios-control/ios-deploy#usage)).

Reply:
Ok, so with your PR we loose this functionality. Which use case did it cover? What will not work any more for users that might be using this right now?

Reply:
I think `extraArgs` is safe to ignore -- this would be any extra arguments to pass `ios-deploy` which we don't really officially support, but was "nice to have". If you do a `cordova run --help` you would see its the `POPTS`, which is not defined, thus we don't have any specific platform options.

We will have to remove any `extraArgs` code here as well: https://github.com/apache/cordova-ios/blob/1d00e60e277e852325e8a3dc867a0416151f9e20/bin/templates/scripts/cordova/lib/run.js#L101-L106